### ISSUE TITLE
equalInAnyOrder

### ISSUE THREAD

**Issue by lpejcic** [oprogramador/deep-equal-in-any-order#63]
When upgraded to the newest version (1.1.14) the equalInAnyOrder method is failing with the error: AssertionError: expected {} to deeply equal {}

Everything works fine if the version is downgraded to the 1.1.13 version

**Comment by micdah** [comment]
The issue seems to occur when an Array contains objects, in which case `new Set(...)` (_which is the primary change_) doesn't produce the same ordering of elements as the previous usage of `sort-any` package did. 

The error message is kind of misleading, as it doesn't show the contents of the arrays as they are now `Set`´s and seems not to be serialized in the log, leading one to think we are comparing two empty objects, when we are actually comparing two non-empty arrays.

I have been able to reproduce this using these simple tests

```javascript
it('array of object fails with new Set', function () {
    const sortDeep = (object) => {
        if (!Array.isArray(object)) {
            if (typeof object !== 'object' || object === null || object instanceof Date) {
                return object;
            }

            return mapValues(object, sortDeep);
        }

        return new Set(object.map(sortDeep));
    }

    const actual = [{name: 'Hello'}, {name: 'World'}]
    const expected = [{name: 'World'}, {name: 'Hello'}]
    expect(sortDeep(actual)).to.deep.equal(sortDeep(expected))
})

it('array of object works with sort-any', function () {
    const sortDeep = (object) => {
        if (!Array.isArray(object)) {
            if (typeof object !== 'object' || object === null || object instanceof Date) {
                return object;
            }

            return mapValues(object, sortDeep);
        }

        return new sortAny(object.map(sortDeep));
    }

    const actual = [{name: 'Hello'}, {name: 'World'}]
    const expected = [{name: 'World'}, {name: 'Hello'}]
    expect(sortDeep(actual)).to.deep.equal(sortDeep(expected))
})
```

**Comment by micdah** [comment]
Created PR with failing test to demonstrate issue: https://github.com/oprogramador/deep-equal-in-any-order/pull/64

**Comment by oprogramador** [comment]
Sorry for this.

I changed logic because all the tests passed.

I want to fix https://github.com/oprogramador/deep-equal-in-any-order/issues/36 and it was the first step.

Maybe I'll fix https://github.com/oprogramador/deep-equal-in-any-order/issues/36 in another way.

**Comment by oprogramador** [comment]
@lpejcic @micdah 

This issue is fixed in the `1.1.15` version.

Thanks, @micdah for your PR.

Btw. now, locally this test `matches same flat arrays of dates in a different order:` fails (though it passed in CI for each env) so I have to fix this. However, it seems to be something random, not related to this fix as the production code was just reverted.

**Comment by oprogramador** [comment]
All the tests are passing locally after running `yarn add sort-any`.
